src/routes_api.py

```diff
@@ -7,19 +7,6 @@
 
 class ApiRoutes:
     def __init__(self, blueprint):
-        @blueprint.route("/api/<cid>/update_roster", methods=["POST"])
-        @auth.require_course_auth
-        @auth.require_admin_status
-        def admin_update_roster(cid):
-            netids = request.json["roster"].strip().lower().split("\n")
-
-            for i, student_id in enumerate(netids):
-                if not util.is_valid_netid(student_id):
-                    return util.error(f"Poorly formatted NetID on line {i + 1}: '{student_id}'")
-
-            db.overwrite_student_roster(cid, netids)
-            return util.success("Successfully updated roster.", HTTPStatus.OK)
-
```

Comment:
I [checked this for usage](https://github.com/search?q=org%3Aillinois-cs241%20update_roster&type=code), and this was only used by an [infrastructure script](https://github.com/illinois-cs241/infrastructure/tree/master/roster-sync) that is broken, so I marked that as deprecated.

Going forward the proper way to update the roster is through `Upload Roster` on on-demand.